nautobot_device_lifecycle_mgmt/api/serializers.py

```diff
@@ -1,52 +1,73 @@
 """API serializers implementation for the LifeCycle Management plugin."""
-from rest_framework import serializers
-
-from nautobot.core.api import SerializedPKRelatedField
-
+from nautobot.core.api import ChoiceField, SerializedPKRelatedField
 from nautobot.dcim.api.nested_serializers import (
     NestedDeviceSerializer,
     NestedDeviceTypeSerializer,
-    NestedPlatformSerializer,
     NestedInventoryItemSerializer,
+    NestedPlatformSerializer,
+)
+from nautobot.extras.api.serializers import (
+    CustomFieldModelSerializer,
+    StatusModelSerializerMixin,
+    StatusSerializerField,
+    TaggedObjectSerializer,
 )
-from nautobot.extras.api.customfields import CustomFieldModelSerializer
-from nautobot.extras.api.serializers import TaggedObjectSerializer, StatusModelSerializerMixin, StatusSerializerField
+from rest_framework import serializers
+
+# Nautobot 1.4 introduced RelationshipModelSerializerMixin
```

Comment:
In 1.4, please use `NautobotModelSerializer` as a base class instead of the pair of (`CustomFieldModelSerializer`, `RelationshipModelSerailizerMixin`).

Reply:
Using `NautobotModelSerializer` triggers a lot of test failures connected to Notes, e.g.

I'd like to add support for Notes in a separate PR.

```
======================================================================
FAIL: test_list_objects_unknown_filter_no_strict_filtering (nautobot_device_lifecycle_mgmt.tests.test_api.VulnerabilityLCMAPITest)
GET a list of objects with an unknown filter parameter and no strict filtering, expect it to be ignored.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/test/utils.py", line 387, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/nautobot/utilities/testing/api.py", line 330, in test_list_objects_unknown_filter_no_strict_filtering
    self.assertHttpStatus(response, status.HTTP_200_OK)
  File "/usr/local/lib/python3.8/site-packages/nautobot/utilities/testing/mixins.py", line 146, in assertHttpStatus
    self.assertIn(response.status_code, expected_status, err_message)
AssertionError: 500 not found in [200] : Expected HTTP status(es) [200]; received 500: {"error": "Reverse for 'vulnerabilitylcm-notes' not found. 'vulnerabilitylcm-notes' is not a valid view function or pattern name.", "exception": "NoReverseMatch", "nautobot_version": "1.4.0", "python_version": "3.8.13"}

======================================================================
FAIL: test_update_object (nautobot_device_lifecycle_mgmt.tests.test_api.VulnerabilityLCMAPITest)
PATCH a single object identified by its ID.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/nautobot/utilities/testing/api.py", line 509, in test_update_object
    self.assertHttpStatus(response, status.HTTP_200_OK)
  File "/usr/local/lib/python3.8/site-packages/nautobot/utilities/testing/mixins.py", line 146, in assertHttpStatus
    self.assertIn(response.status_code, expected_status, err_message)
AssertionError: 500 not found in [200] : Expected HTTP status(es) [200]; received 500: {"error": "Reverse for 'vulnerabilitylcm-notes' not found. 'vulnerabilitylcm-notes' is not a valid view function or pattern name.", "exception": "NoReverseMatch", "nautobot_version": "1.4.0", "python_version": "3.8.13"}

Ran 468 tests in 38.625s

FAILED (failures=62, errors=40, skipped=7)
```

Reply:
Looks like a fix in core might be needed, we shouldn't be throwing a 500 error if a view doesn't support notes. Would you mind filing an issue?

Reply:
Raised https://github.com/nautobot/nautobot/issues/2500